src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -82,6 +82,33 @@ def shrinkLargeFJR(couchDbInstance, sizeLimit):
     return
 
 
+def shrinkLargeSummary(couchDbInstance, sizeLimit):
+    """
+    Look at the CouchDB database queue and reset some potential
+    large fields if they are larger than the max CouchDB limit size.
+    :param couchDbInstance: couchdb database instance
+    :param sizeLimit: integer with the limit number of bytes
+    :return: None
+    """
+    for doc in couchDbInstance._queue:
+        for innerAttr in ["errors", "lumis", "inputfiles"]:
+            fieldBytes = sys.getsizeof(json.dumps(doc[innerAttr]))
+            if fieldBytes > sizeLimit:
+                msg = f"Job summary id: {doc['_id']}, WMBS id: {doc['wmbsid']} "
+                msg += f"has {fieldBytes} bytes under the attr: {innerAttr}. Resetting it.."
+                logging.warning(msg)
+                if isinstance(doc[innerAttr], str):
+                    # keep only the first 5k characters
+                    doc[innerAttr] = doc[innerAttr][:5000]
+                    doc[innerAttr] += "... error message truncated ..."
+                elif isinstance(doc[innerAttr], list):
```

Comment:
Why not (similarly to the str field type) preserve the last, maybe hundreds, of elements from the list?

Reply:
I am going to answer it in the PR body itself to make it more readable.